### ISSUE TITLE
How does the plugin decide on how much of the queue is processed?

### ISSUE THREAD

**Issue by mikkeschiren** [matomo-org/plugin-QueuedTracking#118]
Getting some random behaviors on queue size and items:

```
This worker finished queue processing with 6.05req/s (1340 requests in 221.43 seconds)
Starting to process request sets, this can take a while
This worker finished queue processing with 7.04req/s (55 requests in 7.81 seconds)
Starting to process request sets, this can take a while
This worker finished queue processing with 3.83req/s (201 requests in 52.46 seconds)
```
What decides on how many requests that should be processed? Tried to debug but hard to do without having enough data.

**Comment by tsteur** [comment]
@mikkeschiren you can configure the number of requests that need to be in the queue to start processing in the General Settings UI in Matomo under the Queued Tracking section.

**Comment by mikkeschiren** [comment]
Let me rephrase: My setting for this is 5. So How come I get 1340 requests in one queue and 55 in another? (The total number of requests in the queue at the time were around 25000)

**Comment by tsteur** [comment]
I suppose it depends how often your worker starts for example? If you have 3 queues, and start 3 worker for example every minute, then it should be likely mostly close to zero... Sometimes a worker may finish after a few seconds and exit, then a few requests will pile up (eg 1000) until the next worker starts again... which then processes these 1000 requests again... 

There could be few other reasons that simply more requests somehow are put into a different queue and the worker is not fast enough to catch up etc.

**Comment by mikkeschiren** [comment]
I have 16 workers - they are started every second (16 processes):
```
sh -c while ( php /var/www/html/console queuedtracking:process );do sleep 1; done
```
It does not make sense to me with the random behavior I get. There a random number of requests is processed at a random amount of time. I would really like to control how many requests that are processed every time.

**Comment by mikkeschiren** [comment]
This is the config I am using:

```
[QueuedTracking]
notify_queue_threshold_single_queue = 250000
numQueueWorkers = 16
numRequestsToProcess = 5
processDuringTrackingRequest = 0
queueEnabled = 1
redisDatabase = 0
redisHost = "redis-0,redis-1,redis-2"
redisPassword = somesecretpass
redisPort = "26379,26379,26379"
sentinelMasterName = "redisMaster"
useSentinelBackend = 1
```

**Comment by tsteur** [comment]
It's hard to know what is happening there. A few things I would try:
* Launch workers less often this might only add overhead... maybe start them every 10s or 60s...
* To make things faster etc I recommend adding `--queue-id=0...15` as an option to the process command. so `0` for the first queue, `1` for the second queue etc.

Not sure if this will change anything but worth a try. The logic in the code is otherwise pretty clear as soon as there are X requests in the queue it will start processing it. If you have 16 workers it is possible that some queues get more requests than others but if there is no bottleneck in your system then this likely won't be an issue.

I suppose you are aware of the monitor command and would maybe watch it for a while to maybe understand better what is happening and also monitor your environment. Depending on your users etc it might simply happen that more visitors are put into a specific queue as it depends eg on IP into which queue a request is put
